File: ImaginaryWebshop.Data/Models/WarehouseModel.cs
namespace ImaginaryWebshop.Data.Models
{
    public class WarehouseModel
    {
        [Key] public Guid Id { get; set; }
        [ForeignKey(nameof(Product))] public Guid ProductId { get; set; }
        public ProductModel Product { get; set; } = null!;
        public int StockQuantity { get; set; }
        public string Position { get; set; } = string.Empty;
    }
}

File: ImaginaryWebshop.Web/src/app/models/warehouse.model.ts
export interface WarehouseCreateDto {
    productId: string;
    stockQuantity: number;
    position: string;
}

export interface WarehouseUpdateDto {
    stockQuantity?: number;
    position?: string;
}

export interface WarehouseDetailsDto {
    id: string;
    productId: string;
    stockQuantity: number;
    position: string;
}